pkg/controllers/dnshealthcheckprobe/suite_test.go

```diff
@@ -0,0 +1,142 @@
+	"github.com/Kuadrant/multicluster-gateway-controller/pkg/health"
+	//+kubebuilder:scaffold:imports
+)
+
+// These tests use Ginkgo (BDD-style Go testing framework). Refer to
+// http://onsi.github.io/ginkgo/ to learn more about Ginkgo.
+
+var cfg *rest.Config
+var k8sClient client.Client
+var testEnv *envtest.Environment
+var ctx context.Context
+var cancel context.CancelFunc
+
+func TestAPIs(t *testing.T) {
+	RegisterFailHandler(Fail)
+
+	RunSpecs(t, "Controller Suite")
+}
+
+var _ = BeforeSuite(func() {
+	logf.SetLogger(zap.New(zap.WriteTo(GinkgoWriter), zap.UseDevMode(true)))
+
+	ctx, cancel = context.WithCancel(context.TODO())
+
+	By("bootstraping test environment")
+	testEnv = &envtest.Environment{
+		CRDDirectoryPaths:     []string{filepath.Join("..", "..", "..", "config", "crd", "bases")},
+		ErrorIfCRDPathMissing: true,
+	}
+
+	var err error
+	// cfg is defined in this file globally
+	cfg, err = testEnv.Start()
+	Expect(err).NotTo(HaveOccurred())
+	Expect(cfg).NotTo(BeNil())
+
+	err = v1alpha1.AddToScheme(scheme.Scheme)
+	Expect(err).NotTo(HaveOccurred())
+
+	//+kubebuilder:scaffold:scheme
+
+	k8sClient, err = client.New(cfg, client.Options{Scheme: scheme.Scheme})
+	Expect(err).NotTo(HaveOccurred())
+	Expect(k8sClient).NotTo(BeNil())
+
+	k8sManager, err := ctrl.NewManager(cfg, ctrl.Options{
+		Scheme:                 scheme.Scheme,
+		HealthProbeBindAddress: "0",
+		MetricsBindAddress:     "0",
+	})
+	Expect(err).ToNot(HaveOccurred())
+
+	healthQueue := health.NewRequestQueue(1 * time.Second)
+	err = k8sManager.Add(healthQueue)
+	Expect(err).ToNot(HaveOccurred())
+
+	monitor := health.NewMonitor()
+	err = k8sManager.Add(monitor)
+	Expect(err).ToNot(HaveOccurred())
+
+	healthServer := &testHealthServer{
+		Port: 3333,
+	}
+	err = k8sManager.Add(healthServer)
+	Expect(err).ToNot(HaveOccurred())
+
+	err = (&DNSHealthCheckProbeReconciler{
+		Client:        k8sManager.GetClient(),
+		HealthMonitor: monitor,
+		Queue:         healthQueue,
+	}).SetupWithManager(k8sManager)
+	Expect(err).ToNot(HaveOccurred())
+
+	go func() {
+		defer GinkgoRecover()
+		err = k8sManager.Start(ctx)
+		Expect(err).ToNot(HaveOccurred(), "failed to run manager")
+	}()
+
+})
+
+var _ = AfterSuite(func() {
+	cancel()
+	By("tearing down the test environment")
+	err := testEnv.Stop()
+	Expect(err).NotTo(HaveOccurred())
+})
+
+type testHealthServer struct {
+	Port int
+}
+
+func (s *testHealthServer) Start(ctx context.Context) error {
+	mux := http.NewServeMux()
+
+	endpoint := func(expectedCode int) func(http.ResponseWriter, *http.Request) {
+		return func(w http.ResponseWriter, r *http.Request) {
+			w.WriteHeader(expectedCode)
+		}
+	}
+
+	mux.HandleFunc("/healthy", endpoint(200))
+	mux.HandleFunc("/unhealthy", endpoint(500))
+
+	errCh := make(chan error)
+
+	go func() {
+		errCh <- http.ListenAndServe(fmt.Sprintf(":%d", s.Port), mux)
+	}()
+
+	select {
+	case <-ctx.Done():
+		return ctx.Err()
+	case err := <-errCh:
+		return err
+	}
+}
```

Comment:
All the ginkgo tests should now be in `test/integeration` this won't be run in CI otherwise.